docs/source/docs/registry.md

```diff
@@ -1,32 +1,83 @@
-`deepreg/model/backbone/u_net.py` as an example. Moreover, the corresponding
-`__init__.py` (maybe multiple ones) should be modified so that these classes will be
-automatically registered when executing `import deepreg`. For defining custom classes
-outside DeepReg, more detailed tutorial will be released later.
-
-For now, we only support custom classes of
-
-- backbone
-- loss
+DeepReg adopts the
+[registry system](https://github.com/DeepRegNet/DeepReg/blob/main/deepreg/registry.py)
+to facilitate the addition of custom functionalities.
+
+## Description
+
+The class `Registry` maintains a dictionary mapping `(category, key)` to `value`, where
+
+- `category` is the class category, e.g. `"backbone_class""` for backbone classes.
+- `key` is the name of the registered class, e.g. `"unet"` for the class `UNet`.
+- `value` is the registered class, e.g. `UNet` corresponding to `"unet"`.
+
+A global variable `REGISTRY = Registry()` is defined to provide a central control of all
+classes. The supported categories and the registered classes are resumed in the
+[registered classes](registered_classes.html) page.
+
+### Register a class
+
+To register a class into `REGISTRY`, it is recommended to use `register` as a
+**decorator**. Consider the `UNet` class for backbone as an example.
+
+```python
+@REGISTRY.register(category="backbone_class", name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+The decorator will automatically register the class when this class get imported.
+Therefore, in order to make sure this class registered when we `import deepreg`, this
+class and related parent modules needs to be imported in the `__init__.py` files.
+
+For the purpose of code simplicity, a specific register function is defined for each
+category. For instance, we can use `register_backbone` for `UNet`:
+
+```python
+@REGISTRY.register_backbone(name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+### Instantiate a class
+
+To instantiate a registered example in `REGISTRY`, we call `build_from_config` , which
+allows creating a class instance from a config directly. The config should be a
+dictionary containing the key `name` and other required keys for the class. Please check
+the [configuration documentation](configuration.html) and the docstring of the related
+classes for detailed configuration requirements. The path of the registered classes are
+resumed in the [registered classes](registered_classes.html) page.
+
+For instance, to instantiate a UNet class,
+
+```python
+unet = REGISTRY.build_from_config(category="backbone_class", config={"name":"unet", **kwargs})
+```
+
+where `kwargs` represents other required arguments.
+
+Similarly, for the purpose of code simplicity, a specific build function is defined for
+each category. For instance, we can use `build_backbone` for `UNet`:
+
+```python
+unet = REGISTRY.build_from_config(config={"name":"unet", **kwargs})
```

Comment:
Is this supposed to be `build_backbone`? Change to:
`unet = REGISTRY.build_backbone(config={"name":"unet", **kwargs})`